common/lib/xmodule/xmodule/js/src/video/00_sjson.js

```diff
@@ -27,11 +27,27 @@ function() {
             return sjson.text.length;
         };
 
-        var search = function (time) {
-            var start = getStartTimes(),
+        var search = function (time, startTime, endTime) {
+           var start,
                 max = size() - 1,
                 min = 0,
-                index;
+                index,
+                results;
+
+            /* if we specify a start and end time to search,
+             * search the filtered list of captions in between
+             * the start / end times.
+             * Else, search the unfiltered list.
+             */
+            if (typeof startTime != 'undefined' &&
+                typeof endTime != 'undefined' &&
```

Comment:
Why is the check against `endTime` more thorough than the check on `startTime` (we compare it against both `null` and `undefined`).

Should we default `endTime` to `undefined` for some semantic simplicity? If the user hasn't specified an `endTime`, then it's `undefined`. Else, it's some numerical form.

Similarly, should we force `startTime` to be either a number or undefined? and/or we can coalesce it to zero (`startTime = startTime || 0`).